### ISSUE TITLE
Nametag buffering

### ISSUE THREAD

**Issue by andreasdc** [tr7zw/Exordium#96]
Hey, do you think you can get nametag buffering to work again? Because right now it hides all nametags when enabled, I remember that I really liked that feature when it was working. :)

**Comment by tr7zw** [comment]
As I mentioned on the other issue, probably would rather remove it. I don't see many use cases for it, for how much it changes/lags the nametags vs the benefits. Using Immediately Fast already mostly fixes the nametag-caused lag.

**Comment by andreasdc** [comment]
> As I mentioned on the other issue, probably would rather remove it. I don't see many use cases for it, for how much it changes/lags the nametags vs the benefits. Using Immediately Fast already mostly fixes the nametag-caused lag.

Enabling (right now hiding nametags) gives me 100 fps, so it is really nice boost. I don't remember it was somehow broken when it was working. Only it had full bright, but I played on full bright anyway.

**Comment by tr7zw** [comment]
You are probably talking about the 1.20 version, like the version description says, it's just a "better than nothing" release because people kept asking, even tho I don't have the time right now for it. I don't even remember rather I even tested nametags. As I said, put in Immediately Fast and Sodium, and it will be significantly faster than forcefully breaking the nametag rendering(together with Nvidium I got about 1000fps or something with a 32-chunk view distance, so 🤷‍♂️). Otherwise might want to give Vulkanmod a try, depending on the hardware that might work really well.

**Comment by andreasdc** [comment]
I had nametag buffering on 1.18 with fastergui I think. I have too old card for nvidium and vulkanmod had worse performance than sodium when I tested it on 1.19. If it worked before and it gives massive boost I think it's nice to have it. This feature changed the font a little bit too if I'm not wrong, I can give it a try without mods that remove nametag background etc too.

**Comment by andreasdc** [comment]
Even with just fabric api I don't see nametags. :/

**Comment by tr7zw** [comment]
Just took a look, seems like the nametag overlay is just never rendered. Must have gotten lost during the update.

**Comment by andreasdc** [comment]
Can't wait to test it. :) You have really big and useful mod, if you want some more donations or something you can add tiny premium features or something.

**Comment by tr7zw** [comment]
Been there, tried that, DDoS-sed my own server, and someone thought I was trying to spy on people. All for a solid 0 donations. So yea, not going to try that anymore. 
I'm not 100% sure, but it looks like this also has been broken in 1.19.4 already without anyone noticing. 12 am is just not the time to look into this mess, if I try to add it like it was in 1.19.3, then it seems to render the nametag in the skybox or something. Will look into it at some later point.

**Comment by andreasdc** [comment]
Keep doing the good work and don't look ad id*ots xD For example if you would do extra checks to handle bossbar, title etc. for premium options for any donation like $1 - you would have them for sure and it wouldn't hurt anyone too. Sure thing, we can talk later about nametags, I wasn't using it too because it was broken, but I remember it gave nice boost.

**Comment by RedGreenBlue09** [comment]
I think this issue should be prioritized as nametag rendering takes as much as 15% of the rendering time.

**Comment by andreasdc** [comment]
It is true.

**Comment by tr7zw** [comment]
Please use ImmediatelyFast. I'll remove all the sign buffering and name tag buffering logic with the next release probably.

**Comment by andreasdc** [comment]
We are talking with ImmediatelyFast already, at least I am talking with this. It is not enough and nametag rendering takes huge amount of fps, you have around 30% of what you would actually have. Did you see what nametags do on Hypixel lobby for example?

**Comment by tr7zw** [comment]
Yes I did all my testing in the hypixel lobbies. But to make it short, I don't have the time for that code, it doesn't match the premise of the mod, no one used it, it looked bad. With IF now having a better version of the sign buffering, might as well take this opportunity to nuke all hud unrelated code.

**Comment by andreasdc** [comment]
Sign buffering maybe I can believe that noone used it. But do you really think you don't see nametags in Minecraft, like 99.9% of the time in multiplayer? Something that takes huge amout on fps?
Maybe someone would help you like with the tablist buffering, but leaving it is tragic, it won't be added anywhere else.

**Comment by tr7zw** [comment]
The sign buffering is turned on by default, so that's used by nearly everyone. I just remove it since there is a better one now. And if anyone wants to pick up the name tag feature, feel free to do so.

**Comment by andreasdc** [comment]
I think it's better to keep it somehow or find someone that can finish it, it was working well and the boost was massive.

**Comment by RedGreenBlue09** [comment]
I think it is fine to remove Nametag buffering in the next release as it's already broken. But please keep this issue open until maybe this mod or IF implemented it.

**Comment by andreasdc** [comment]
Isn't it better to fix it? What is so hard there compared to the massive improvement?

**Comment by RedGreenBlue09** [comment]
But he can't fix it right now, so just remove that broken feature. Maybe it'll comeback in the future.

**Comment by tr7zw** [comment]
> Isn't it better to fix it? What is so hard there compared to the massive improvement?

I already gave it a shot, and it wasn't some super low-hanging fix. And I don't see this feature as a massive improvement. You have to remember that I make these mods for fun as a hobby. This feature is annoying to fix, I would never use it myself and I don't think the end result when it works is good, hence I'm not interested in investing potentially hours to fix it(especially since I just don't have time right now due to IRL stuff). If anything, I'd try to make a better solution at some later point in time(custom shaders, as I now know how to create one for mc, or maybe using the dynamic texture atlas solution that IF implemented for their sign buffering logic(that's also where the speedup compared to my implementation comes from)).

**Comment by RedGreenBlue09** [comment]
> > Isn't it better to fix it? What is so hard there compared to the massive improvement?
> 
> I already gave it a shot, and it wasn't some super low-hanging fix. And I don't see this feature as a massive improvement. You have to remember that I make these mods for fun as a hobby. This feature is annoying to fix, I would never use it myself and I don't think the end result when it works is good, hence I'm not interested in investing potentially hours to fix it(especially since I just don't have time right now due to IRL stuff). If anything, I'd try to make a better solution at some later point in time(custom shaders, as I now know how to create one for mc, or maybe using the dynamic texture atlas solution that IF implemented for their sign buffering logic(that's also where the speedup compared to my implementation comes from)).

I fully agree on this.

**Comment by RedGreenBlue09** [comment]
It is totally not forgotten lol.

> If anything, I'd try to make a better solution at some later point in time

**Comment by andreasdc** [comment]
Any plans on fixing this feature? It would be so good
